### PR TITLE
Invalid date when timeOnly defined on datetime UIComponent.

### LINKED ISSUE #23157: Invalid date when timeOnly defined on datetime UIComponent
### Preconditions (*)
1. Magento 2.3.1

### Steps to reproduce (*)
1. Define a ui_component form.xml
1. add a Datetime UI component with the option 'timeOnly' set to 'yes'
```
<field name="start_time"]
    [argument name="data" xsi:type="array"]
        [item name="config" xsi:type="array"]
            [item name="dataType" xsi:type="string"]string[/item]
            [item name="label" xsi:type="string" translate="true"]Start Time[/item]
            [item name="formElement" xsi:type="string"]date[/item]
            [item name="source" xsi:type="string"]import_profile[/item]
            [item name="dataScope" xsi:type="string"]start_time[/item]
            [item name="options" xsi:type="array"]
                [item name="timeOnlyTitle" xsi:type="string"]Select Start Time[/item]
                [item name="showsTime" xsi:type="boolean"]true[/item]
                [item name="timeOnly" xsi:type="boolean"]true[/item]
                [item name="timeFormat" xsi:type="string"]HH:mm:ss[/item]
                [item name="dateFormat" xsi:type="string"]yyyy-MM-dd[/item]
            [/item]
        [/item]
    [/argument]
[/field>
{code}
1. Save the form and check the result. 

### Expected result (*)
1. Correct time value

### Actual result (*)
1. The value states: 'Invalid Date'

### Additional information
This problem doesn't occur when the option 'timeOnly' is not defined. 

My best guess is that the problem happens inside the /module-ui/view/base/web/js/form/element/date.js at 'onShiftedValueChange' function..

{code}
 /**
        * Prepares and sets date/time value that will be sent
        * to the server.
        * * @param {String} shiftedValue
         */
        onShiftedValueChange: function (shiftedValue) {
            var value,
                formattedValue,
                momentValue;

            if (shiftedValue) {
                momentValue = moment(shiftedValue, this.pickerDateTimeFormat);

                if (this.options.showsTime) {
                    formattedValue = moment(momentValue).format(this.timezoneFormat);
                    //formattedValue returns an incorrect value. 
                    value = moment.tz(formattedValue, this.storeTimeZone).tz('UTC').toISOString();
                } else {
                    value = momentValue.format(this.outputDateFormat);
                }
            } else {
                value = '';
            }

            if (value !== this.value()) {
                this.value(value);
            }
        },
```

### PR DESCRIPTION
**Pull request by engcom-Kilo**
### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/23157

### Manual testing scenarios (*)

1. see https://github.com/magento/magento2/issues/23157

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

File: app/code/Magento/Ui/view/base/web/js/form/element/date.js [hunk 1]
```diff
@@ -177,9 +177,9 @@ define([
          * with moment.js library.
          */
         prepareDateTimeFormats: function () {
-            this.pickerDateTimeFormat = this.options.dateFormat;
+            this.pickerDateTimeFormat = this.options.timeOnly ? this.options.timeFormat : this.options.dateFormat;
```
Comment (zakdma) [inline app/code/Magento/Ui/view/base/web/js/form/element/date.js]: Could you rewrite L180-L184 to make it more readable
```
if (this.options.timeOnly) {
    this.pickerDateTimeFormat = this.options.timeFormat;
} else {
    this.pickerDateTimeFormat = this.options.dateFormat;
    if (this.options.showsTime) {
        this.pickerDateTimeFormat += ' ' + this.options.timeFormat;
    }
}
```

**Review by zakdma — CHANGES_REQUESTED** [review]

File: app/code/Magento/Ui/view/base/web/js/form/element/date.js (hunk 1 above)
Comment (engcom-Kilo) [reply app/code/Magento/Ui/view/base/web/js/form/element/date.js]: Fixed.

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Review by zakdma — APPROVED** [review]

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Manual testing scenario:**

1. Define a ui_component form.xml
2. Add a Datetime UI component with the option '_**timeOnly**_' set to '**_true_**'

```xml
        <field name="start_date">
            <argument name="data" xsi:type="array">
                <item name="config" xsi:type="array">
                    <item name="dataType" xsi:type="string">string</item>
                    <item name="label" xsi:type="string" translate="true">Start Time</item>
                    <item name="formElement" xsi:type="string">date</item>
                    <item name="source" xsi:type="string">import_profile</item>
                    <item name="dataScope" xsi:type="string">start_time</item>
                    <item name="options" xsi:type="array">
                    <item name="timeOnlyTitle" xsi:type="string">Select Start Time</item>
                    <item name="showsTime" xsi:type="boolean">true</item>
                    <item name="timeOnly" xsi:type="boolean">true</item>
                    <item name="timeFormat" xsi:type="string">HH:mm:ss</item>
                    <item name="dateFormat" xsi:type="string">yyyy-MM-dd</item>
                    </item>
                </item>
            </argument>
        </field>
```
3. Go to Admin -> Content -> Blocks -> Add New Block;
4. Select **Start Time** date;

**Before:** :heavy_multiplication_x: The value states: 'Invalid Date'



**After:** :heavy_check_mark: Correct time value




Everything works well if:

```xml
<item name="timeOnly" xsi:type="boolean">false</item>
```
**Actual Result:** :heavy_check_mark: 

Screenshot from 2021-01-12 12-07-24

and if:

```xml
<item name="showsTime" xsi:type="boolean">false</item>
```

**Actual Result:** :heavy_check_mark: 
Screenshot from 2021-01-12 12-13-03

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma, an error occurred during the Pull Request import.

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma, an error occurred during the Pull Request import.

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-31549`.

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-31549`.

**Comment by zakdma** [comment]
@magento run all tests

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-31549`.

**Pull request merged**